scripts/mbedtls_dev/test_generation.py

```diff
@@ -165,15 +179,19 @@ def generate_target(self, name: str, *target_args) -> None:
         test_cases = self.targets[name](*target_args)
         self.write_test_data_file(name, test_cases)
 
-def main(args, generator_class: Type[TestGenerator] = TestGenerator):
+def main(args, description: str, generator_class: Type[TestGenerator] = TestGenerator):
     """Command line entry point."""
-    parser = argparse.ArgumentParser(description=__doc__)
+    parser = argparse.ArgumentParser(description=description)
     parser.add_argument('--list', action='store_true',
                         help='List available targets and exit')
     parser.add_argument('--list-for-cmake', action='store_true',
                         help='Print \';\'-separated list of available targets and exit')
-    parser.add_argument('--directory', default="tests/suites", metavar='DIR',
+    parser.add_argument('--directory', metavar='DIR',
                         help='Output directory (default: tests/suites)')
+    # The `--directory` option is interpreted relative to the directory from
+    # which the script is invoked, but the default is relative to the root of
+    # the mbedtls tree. The default should not be set above, but instead after
+    # `build_tree.chdir_to_root()` is called.
```

Comment:
Well, not quite. `--directory` is still interpreted relative to the root of the mbedtls tree. An `abspath` call is missing somewhere (or alternatively we could stop using `chdir`, but that would be annoying).

But since this is preexisting (I know it worked when I originally wrote it, but I might have broken it before I even committed), this is a non-blocker.